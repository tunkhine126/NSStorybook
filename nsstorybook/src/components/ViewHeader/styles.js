import { makeStyles } from '@material-ui/core/styles';

export const viewHeaderStyles = makeStyles(() => ({
  header: {
    fontSize: '28px',
  },
}));
