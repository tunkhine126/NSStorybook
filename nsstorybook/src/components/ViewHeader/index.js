/**
 *
 * ViewHeader
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Helmet } from 'react-helmet';

import Typography from '@material-ui/core/Typography';
import { viewHeaderStyles } from './styles';

function ViewHeader({ component, body }) {
  const classes = viewHeaderStyles();

  return (
    <>
      <Helmet>
        <title>{`New Story - ${component}`}</title>
        <meta name="description" content={component} />
      </Helmet>
      <Typography
        variant="h1"
        className={classes.header}
        component="h2"
        gutterBottom
      >
        {body}
      </Typography>
    </>
  );
}

ViewHeader.propTypes = {
  component: PropTypes.string.isRequired,
  body: PropTypes.oneOfType([PropTypes.element, PropTypes.string]).isRequired,
};

export default ViewHeader;
