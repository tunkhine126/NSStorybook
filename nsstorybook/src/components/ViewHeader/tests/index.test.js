/**
 *
 * Tests for ViewHeader
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from '@testing-library/react';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import ViewHeader from '../index';

describe('<ViewHeader />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<ViewHeader component="Test" body={<div>New Component</div>} />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<ViewHeader component="Test" body={<div>New Component</div>} />);
    expect(firstChild).toMatchSnapshot();
  });
});
