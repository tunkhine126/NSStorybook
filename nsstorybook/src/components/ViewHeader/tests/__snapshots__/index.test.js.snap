// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewHeader /> Should render and match the snapshot 1`] = `
<h2
  class="MuiTypography-root makeStyles-header-32 MuiTypography-h1 MuiTypography-gutterBottom"
>
  <div>
    New Component
  </div>
</h2>
`;
